web/views.py

```diff
@@ -67,4 +68,29 @@ def register(request):
     else:
       return render(request, 'register.html', {
         'form': form,
-      })
\ No newline at end of file
+      })
+
+def logout_view(request):
+  logout(request)
+  return redirect('/')
+
+def login_user(request):
+  if request.method == 'GET':
+    return render(request, 'login.html')
+  else:
+    username = request.POST['username']
+    password = request.POST['password']
+    errors=[]
+    user_auth = authenticate(username=username, password=password)
+    if user_auth is not None:
+        if user_auth.is_active:
```

Comment:
Make sure your indentation is consistent with the rest of the file. I would actually prefer to move all the code to 4-space indentation, but we should probably save that for another revision.